build-tools/packages/build-cli/src/commands/report/codeCoverage.ts

```diff
@@ -0,0 +1,166 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Flags } from "@oclif/core";
+import { getCodeCoverageReport } from "../../codeCoverage/codeCoveragePr.js";
+import {
+	getPackagesWithCodeCoverageChanges,
+	isCodeCoverageCriteriaPassed,
+} from "../../codeCoverage/compareCodeCoverage.js";
+import { getCommentForCodeCoverageDiff } from "../../codeCoverage/getCommentForCodeCoverage.js";
+import { type IAzureDevopsBuildCoverageConstants } from "../../library/azureDevops/constants.js";
+import { createOrUpdateCommentOnPr, getChangedFilePaths } from "../../library/githubRest.js";
+import { BaseCommand } from "../../library/index.js";
+
+// Unique identifier for the comment made on the PR. This is used to identify the comment
+// and update it based on a new build.
+const commentIdentifier = `<!-- Code coverage automated comment -->`;
+
+export default class ReportCodeCoverageCommand extends BaseCommand<
+	typeof ReportCodeCoverageCommand
+> {
+	static readonly description = "Run comparison of code coverage stats";
+
+	static readonly flags = {
+		ADO_BUILD_ID: Flags.integer({
```

Comment:
nit: typically we use standard variable naming conventions for the flag names. E.g. adoBuildId. You can/should still use uppercase snake case for the env variables like you have. But the flag names don't have to match, and since you use the flag names in the code, camelCase can make it more readable.